httpx/_auth.py

```diff
@@ -280,7 +280,7 @@ def digest(data: bytes) -> bytes:
 
         qop = self._resolve_qop(challenge.qop, request=request)
         if qop is None:
-            digest_data = [HA1, challenge.nonce, HA2]
+            digest_data = [challenge.nonce, HA2]
         else:
             digest_data = [challenge.nonce, nc_value, cnonce, qop, HA2]
```

Comment:
Looking at the curl code, seems like this change should be the other way around(?)...

```python
        if qop is None:
            digest_data = [HA1, challenge.nonce, HA2]
        else:
            digest_data = [HA1, challenge.nonce, nc_value, cnonce, qop, HA2]
```

And then...

```python
        format_args = {
            "username": self._username,
            "realm": challenge.realm,
            "nonce": challenge.nonce,
            "uri": path,
            "response": digest(key_digest),
            "algorithm": challenge.algorithm.encode(),
        }
```

Reply:
Yeah, initially I wanted to avoid touching the other branch (so I don't break it accidentally), but I agree the curl way is more readable. I also add the comments.